src/main/java/com/drajer/bsa/kar/action/SubmitReport.java

```diff
@@ -50,59 +50,73 @@ public BsaActionStatus process(KarProcessingData data, EhrQueryService ehrServic
           resourcesToSubmit.addAll(resources);
         }
       }
+      
+      Set<UriType> endpoints = data.getKar().getReceiverAddresses();
+      if (endpoints.size() > 0) {
+        for (UriType uri : endpoints) {
+          submitResources(resourcesToSubmit, data, ehrService, actStatus, uri.getValueAsString());
+        }
+      }
+      if(!submissionEndpoint.isEmpty()){
```

Comment:
So this is something we may want to look further into. I would assume that if the submissionEndpoint (which is the one that comes from the properties file and I think refers to a TTP) was provided, then we wouldn't send to the defined endpoints directly, because the TTP would be responsible for forwarding them along to the appropriate location. But I don't fully understand the real world use cases here since I only have the MedMorph sequence diagrams to work from